packages/browser/src/ClientAuthentication.ts

```diff
@@ -125,6 +112,10 @@ export default class ClientAuthentication {
   handleIncomingRedirect = async (
     url: string
   ): Promise<ISessionInfo | undefined> => {
-    return this.redirectHandler.handle(url);
+    const redirectInfo = await this.redirectHandler.handle(url);
+    // TODO: When handling DPoP, both the key and the token should be returned
+    // by the redirect handler.
+    this.fetch = buildBearerFetch(redirectInfo.accessToken);
+    return { ...redirectInfo } as ISessionInfo;
```

Comment:
Is this not the same, or do you explicitly want a clone? And is the type assertion really necessary?

```suggestion
    return redirectInfo as ISessionInfo;
```

Reply:
That was actually laziness from my part: I did not want the `accessToken` to be returned visibly, hence the type assertion. But now that I think about it cloning the `redirectInfo` object would not prevent the property to be present in the instance, even if it is hidden in the type. I fixed this properly in cf94fc1e3c6ca8ad2aaca8b92b57a840da9cc224.